### ISSUE TITLE
v3.2.4 Daemon step in running state, but dependents steps in child template don't start

### ISSUE THREAD

**Issue by jccarte** [argoproj/argo-workflows#7408]
## Summary

What happened/what you expected to happen?
When updating to 3.2.4 workflowtemplaterefs containing a steps template with a deamon step do not trigger subsequent steps. Seems to be a similar issue to https://github.com/argoproj/argo-workflows/issues/6379 but with steps templates triggered from within another workflow template. I have created a generic example that works on 3.1.15 and not on 3.2.4. In this example the do-something step works fine, and then the postgres daemon step starts up, but the "run-tests" is never created and it just spins forever.

What version is it broken in?
3.2.X (tested in 3.2.4)
What version was it working in?
3.1.15
## Diagnostics

Either a workflow that reproduces the bug, or paste you whole workflow YAML, including status, something like:
### Top level workflow template:
```yaml
apiVersion: argoproj.io/v1alpha1
kind: WorkflowTemplate
metadata:
  name: broken-pipeline
spec:
  entrypoint: main
  volumes:
    - name: workspace
      emptyDir: {}
  templates:
  - name: do-something
    container:
      image: alpine:3.7
      command: [echo, "do something"]
  - name: main
    dag:
      tasks:
        - name: do-something
          template: do-something
        - name: run-tests-broken
          depends: "do-something"
          templateRef:
            name: run-tests-broken
            template: main
```
### Child workflow template:
```yaml
apiVersion: argoproj.io/v1alpha1
kind: WorkflowTemplate
metadata:
  name: run-tests-broken
spec:
  entrypoint: main
  templates:
  - name: main
    steps:
      - - name: postgres
          template: postgres
          arguments:
            parameters:
              - name: postgres-image-tag
                value: "13"
      - - name: run-tests-broken
          template: run-tests-broken
  - name: run-tests-broken
    container:
      image: alpine:3.7
      command: [echo, "run-tests"]
  - name: postgres
    inputs:
      parameters:
        - name: postgres-image-tag
    daemon: true
    container:
      image: "postgres:{{inputs.parameters.postgres-image-tag}}"
      name: database
      env:
        - name: POSTGRES_PASSWORD
          value: postgres
 ```

What Kubernetes provider are you using?
GKE
What executor are you running? Docker/K8SAPI/Kubelet/PNS/Emissary
PNS
```bash
# Logs from the workflow controller:time="2021-12-14T16:23:31.420Z" level=info msg="Processing workflow" namespace=argo-workflows workflow=broken-pipeline-gzwmq
time="2021-12-14T16:23:31.437Z" level=info msg="Updated phase  -> Running" namespace=argo-workflows workflow=broken-pipeline-gzwmq
time="2021-12-14T16:23:31.437Z" level=info msg="DAG node broken-pipeline-gzwmq initialized Running" namespace=argo-workflows workflow=broken-pipeline-gzwmq
time="2021-12-14T16:23:31.438Z" level=info msg="All of node broken-pipeline-gzwmq.do-something dependencies [] completed" namespace=argo-workflows workflow=broken-pipeline-gzwmq
time="2021-12-14T16:23:31.438Z" level=info msg="Pod node broken-pipeline-gzwmq-3662255146 initialized Pending" namespace=argo-workflows workflow=broken-pipeline-gzwmq
time="2021-12-14T16:23:31.483Z" level=info msg="Created pod: broken-pipeline-gzwmq.do-something (broken-pipeline-gzwmq-3662255146)" namespace=argo-workflows workflow=broken-pipeline-gzwmq
time="2021-12-14T16:23:31.483Z" level=info msg="TaskSet Reconciliation" namespace=argo-workflows workflow=broken-pipeline-gzwmq
time="2021-12-14T16:23:31.483Z" level=info msg=reconcileAgentPod namespace=argo-workflows workflow=broken-pipeline-gzwmq
time="2021-12-14T16:23:31.493Z" level=info msg="Workflow update successful" namespace=argo-workflows phase=Running resourceVersion=1868766644 workflow=broken-pipeline-gzwmq
time="2021-12-14T16:23:41.484Z" level=info msg="Processing workflow" namespace=argo-workflows workflow=broken-pipeline-gzwmq
time="2021-12-14T16:23:41.485Z" level=info msg="Updating node broken-pipeline-gzwmq-3662255146 exit code 0" namespace=argo-workflows workflow=broken-pipeline-gzwmq
time="2021-12-14T16:23:41.485Z" level=info msg="Setting node broken-pipeline-gzwmq-3662255146 outputs: {\"artifacts\":[{\"name\":\"main-logs\",\"gcs\":{\"key\":\"broken-pipeline-gzwmq/broken-pipeline-gzwmq-3662255146/main.log\"}}]}" namespace=argo-workflows workflow=broken-pipeline-gzwmq
time="2021-12-14T16:23:41.485Z" level=info msg="Updating node broken-pipeline-gzwmq-3662255146 status Pending -> Succeeded" namespace=argo-workflows workflow=broken-pipeline-gzwmq
time="2021-12-14T16:23:41.485Z" level=info msg="All of node broken-pipeline-gzwmq.run-tests-broken dependencies [do-something] completed" namespace=argo-workflows workflow=broken-pipeline-gzwmq
time="2021-12-14T16:23:41.490Z" level=info msg="Steps node broken-pipeline-gzwmq-863504055 initialized Running" namespace=argo-workflows workflow=broken-pipeline-gzwmq
time="2021-12-14T16:23:41.490Z" level=info msg="StepGroup node broken-pipeline-gzwmq-645885843 initialized Running" namespace=argo-workflows workflow=broken-pipeline-gzwmq
time="2021-12-14T16:23:41.491Z" level=info msg="Pod node broken-pipeline-gzwmq-2252754156 initialized Pending" namespace=argo-workflows workflow=broken-pipeline-gzwmq
time="2021-12-14T16:23:41.514Z" level=info msg="Created pod: broken-pipeline-gzwmq.run-tests-broken[0].postgres (broken-pipeline-gzwmq-2252754156)" namespace=argo-workflows workflow=broken-pipeline-gzwmq
time="2021-12-14T16:23:41.514Z" level=info msg="Workflow step group node broken-pipeline-gzwmq-645885843 not yet completed" namespace=argo-workflows workflow=broken-pipeline-gzwmq
time="2021-12-14T16:23:41.514Z" level=info msg="TaskSet Reconciliation" namespace=argo-workflows workflow=broken-pipeline-gzwmq
time="2021-12-14T16:23:41.514Z" level=info msg=reconcileAgentPod namespace=argo-workflows workflow=broken-pipeline-gzwmq
time="2021-12-14T16:23:41.526Z" level=info msg="Workflow update successful" namespace=argo-workflows phase=Running resourceVersion=1868767282 workflow=broken-pipeline-gzwmq
time="2021-12-14T16:23:51.516Z" level=info msg="Processing workflow" namespace=argo-workflows workflow=broken-pipeline-gzwmq
time="2021-12-14T16:23:51.516Z" level=info msg="Updating node broken-pipeline-gzwmq-2252754156 status Pending -> Running" namespace=argo-workflows workflow=broken-pipeline-gzwmq
time="2021-12-14T16:23:51.517Z" level=info msg="Workflow step group node broken-pipeline-gzwmq-645885843 not yet completed" namespace=argo-workflows workflow=broken-pipeline-gzwmq
time="2021-12-14T16:23:51.517Z" level=info msg="TaskSet Reconciliation" namespace=argo-workflows workflow=broken-pipeline-gzwmq
time="2021-12-14T16:23:51.517Z" level=info msg=reconcileAgentPod namespace=argo-workflows workflow=broken-pipeline-gzwmq
time="2021-12-14T16:23:51.530Z" level=info msg="Workflow update successful" namespace=argo-workflows phase=Running resourceVersion=1868767766 workflow=broken-pipeline-gzwmq
time="2021-12-14T16:24:01.531Z" level=info msg="Processing workflow" namespace=argo-workflows workflow=broken-pipeline-gzwmq
time="2021-12-14T16:24:01.532Z" level=info msg="Workflow step group node broken-pipeline-gzwmq-645885843 not yet completed" namespace=argo-workflows workflow=broken-pipeline-gzwmq
time="2021-12-14T16:24:01.532Z" level=info msg="TaskSet Reconciliation" namespace=argo-workflows workflow=broken-pipeline-gzwmq
time="2021-12-14T16:24:01.532Z" level=info msg=reconcileAgentPod namespace=argo-workflows workflow=broken-pipeline-gzwmq
```

---

**Message from the maintainers**:

Impacted by this regression? Give it a 👍. We prioritise the issues with the most 👍.

**Comment by alexec** [comment]
I wonder if could have the same cause as #7388 .

**Comment by alexec** [comment]
9d3c9809c chore: faster `make start`: no API by default (#7146) ** PASS **
0ee774315 docs: Add link to awesome-argo for more resources (#7166) ** FAIL **

**Comment by alexec** [comment]
Should be fixed on `latest` and will be on v3.2.7
